libraries/provider_pg_hba.rb

```diff
@@ -0,0 +1,25 @@
+class Chef
+  class Provider
+    # builds an array of slave systems for use in pg_hba.conf file
+    class PgHbaConfig < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      action :create do
+        new_resource.slave_ips.each do |slaveip|
+          node.default['postgresql']['pg_hba'] << {
```

Comment:
I'm wondering if setting default attributes at converge time instead of compile time is going to surprise someone. Luckily, the upstream cookbook doesn't pass variables for the pg_hba.conf template -- it evaluates them at the time the template is rendered.